django/forms/models.py

```diff
@@ -279,6 +279,13 @@ def __new__(mcs, name, bases, attrs):
                 # fields from the model"
                 opts.fields = None
 
+            if opts.exclude:
```

Comment:
Alternatively, you could handle this where `fields` is updated with `new_class.declared fields` like so:

```
for field_name, field in new_class.declared_fields.iteritems():
    if opts.exclude and field_name in opts.exclude:
        continue
    if opts.fields and field_name not in opts.fields:
        continue
    fields[field_name] = field
```